log4j-core/src/main/java/org/apache/logging/log4j/core/util/internal/instant/InstantPatternFormatter.java

```diff
@@ -28,6 +28,12 @@
 
 /**
  * Contract for formatting {@link Instant}s using a date & time formatting pattern.
+ * <h2>Internal usage only!</h2>
+ * <p>
+ * This class is intended only for internal Log4j usage.
+ * <b>Log4j users should not use this class!</b>
+ * This class is not subject to any backward compatibility concerns.
+ * </p>
```

Comment:
Do we really need access to this class from JTL?
Can't we just instantiate all the formatters from an `InstantFormatter` static factory method?

We could have for example an `InstantFormatter.newInstance(String pattern, TimeZone, Locale)` public method. I don't see why we would need a builder pattern: these 3 parameters were enough for a very long time. Of course, `TimeZone` will be ignored by `InstantNumberFormatter`, but `Locale` might apply to both formatters (if users request it of course).

Reply:
> Do we really need access to this class from JTL? Can't we just instantiate all the formatters from an `InstantFormatter` static factory method?
> 
> We could have for example an `InstantFormatter.newInstance(String pattern, TimeZone, Locale)` public method. I don't see why we would need a builder pattern: these 3 parameters were enough for a very long time. Of course, `TimeZone` will be ignored by `InstantNumberFormatter`, but `Locale` might apply to both formatters (if users request it of course).

Why shall we sacrifice from type safety to avoid exporting two classes, where we explicitly stated they are intended for internal usage only? If this approach would have helped to avoid exporting the entire package, I could have agreed with the sentiment, but that is not the case.